### PR TITLE
fix memory leaks found by address sanitizer and enable even more tests

### PR DESCRIPTION
**Pull request by jluebbe**

### CODE REVIEW

File: test/common.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #include "manifest.h"
 
-#define memdup(x) (g_memdup(x, sizeof(*x)))
```
Comment (ejoerns) [inline test/common.h:7]: This seems to break the test suite (test/nbd-test).

Comment (jluebbe) [reply test/common.h:7]: Should be fixed by the new approach.

File: test/common.c [hunk 2]
```diff
@@ -408,3 +408,26 @@ void replace_strdup(gchar **dst, const gchar *src)
 	g_free(*dst);
 	*dst = g_strdup(src);
 }
+
+void* dup_test_mem(GPtrArray *ptrs, const void *mem, gsize len)
```
Comment (ejoerns) [inline test/common.c:412]: Maybe it should somehow reflect that something is added to the list of 'to free' candidates? But I also don't really have a good name that covers both the copying and the freeing aspect...

**Review by ejoerns — COMMENTED** [review]
Looks really good overall! Just some minor comments.

File: test/bundle.c [hunk 3]
```diff
@@ -67,17 +67,17 @@ static void prepare_bundle(BundleFixture *fixture, gconstpointer user_data)
 static void bundle_fixture_set_up_bundle(BundleFixture *fixture,
 		gconstpointer user_data)
 {
-	r_context_conf()->certpath = g_strdup("test/openssl-ca/dev/autobuilder-1.cert.pem");
-	r_context_conf()->keypath = g_strdup("test/openssl-ca/dev/private/autobuilder-1.pem");
+	replace_strdup(&r_context_conf()->certpath, "test/openssl-ca/dev/autobuilder-1.cert.pem");
+	replace_strdup(&r_context_conf()->keypath, "test/openssl-ca/dev/private/autobuilder-1.pem");
```
[the remark this answers is not in the document]
Comment (jluebbe) [reply test/bundle.c:71]: I'd like to do that refactoring separately, as it needs a much more thorough review (compared to these simple changes).

File: test/common.c (hunk 2 above)
Comment (jluebbe) [reply test/common.c:412]: Yes, I've thought about that for some time as well and this is the best I came up with. Can we keep it for now?

**Comment by jluebbe** [comment]
I've force pushed this PR. Only the commit order, separation and messages have changed, so the diff is empty.

**Comment by jluebbe** [comment]
I've rebased on master to ensure no new leaks have been introduced in the meantime. There are no changes to the commits in this PR.

File: test/bundle.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (ejoerns) [reply test/bundle.c:71]: Ok. In order to unblock follow-up work (and as it already allows us to enable more sanity tests), I guess we can merge it like it is first of all.

**Review by ejoerns — APPROVED** [review]

**Pull request merged**